controllers/storagecluster/cephcluster.go

```diff
@@ -694,6 +697,10 @@ func (r *StorageClusterReconciler) checkTuneStorageDevices(ds ocsv1.StorageDevic
 			continue
 		}
 
+		if dt.storageAccountType != storageClass.Parameters["storageaccounttype"] {
+			continue
+		}
+
```

Comment:
Part of the problem here is that you're checking for a parameter that doesn't exist. When you try to fetch a value in a map, Go actually returns two values, the value itself and a boolean. The boolean says whether or not the value was found in the map. Thus, with the refactor I suggested, it would look like this:
```
	for _, dt := range knownDiskTypes {
		if string(dt.provisioner) != storageClass.Provisioner {
			continue
		}

		invalidParameters := false
		for parameter, expectedValue := range dt.StorageClassParameters {
			actualValue, ok := storageClass.Parameters[parameter]
			if ok && actualValue != expectedValue {
				invalidParameters = true
			}
		}

		if invalidParameters {
			continue
		}
		return dt.speed, nil
	}
```